### ISSUE TITLE
internal ssh communicator unable to connect to openssh-8.8+ hosts. 

### ISSUE THREAD

**Issue by AntonioMeireles** [hashicorp/vagrant#12533]
Hi again, 

On a test guest that has **OpenSSH-8.8** inside I'm getting (with latest **2.2.18**)  the following failure:

```WARN -- net.ssh.transport.algorithms[e7a4]: unsupported algorithm: `["rsa-sha2-256", "rsa-sha2-512"]'```
... so provisioning fails... 

In the past we got a similar issue (#12344, #12408, #12381) that went closed by #12415, so at sight,  it seems that [vagrant/lib/vagrant/patches/net-ssh.rb](vagrant/lib/vagrant/patches/net-ssh.rb) needs to be massaged a bit, again, to tackle this.

full error context bellow 
```
INFO ssh:   - Key Path: ["/Users/am/.vagrant.d/insecure_private_key"]
DEBUG ssh:   - connect_opts: {:auth_methods=>["none", "hostbased", "publickey"], :config=>false, :forward_agent=>false, :send_env=>false, :keys_only=>true, :verify_host_key=>:never, :password=>nil, :port=>2222, :timeout=>15, :user_known_hosts_file=>[], :verbose=>:debug, :logger=>#<Logger:0x00007f928d8c5880 @level=0, @progname=nil, @default_formatter=#<Logger::Formatter:0x00007f928d8c5858 @datetime_format=nil>, @formatter=nil, @logdev=#<Logger::LogDevice:0x00007f928d8c5808 @shift_period_suffix=nil, @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<StringIO:0x00007f928d8c58d0>, @binmode=false, @mon_data=#<Monitor:0x00007f928d8c57e0>, @mon_data_owner_object_id=59240>>, :keys=>["/Users/am/.vagrant.d/insecure_private_key"], :remote_user=>"clear"}
DEBUG global: ssh server version detected: 8.8
DEBUG global: ssh server supports deprecation of RSA SHA1, deprecating
DEBUG global: set RSA SHA1 deprecation on public key: bb:e4:5a:cb:47:24:59:5e:27:4a:2a:5e:72:2b:06:61
DEBUG global: set RSA SHA1 deprecation on private key: bb:e4:5a:cb:47:24:59:5e:27:4a:2a:5e:72:2b:06:61
DEBUG ssh: == Net-SSH connection debug-level log START ==
DEBUG ssh: D, [2021-10-06T12:47:35.897252 #68218] DEBUG -- net.ssh.transport.session[e77c]: establishing connection to 127.0.0.1:2222
D, [2021-10-06T12:47:35.897505 #68218] DEBUG -- net.ssh.transport.session[e77c]: connection established
I, [2021-10-06T12:47:35.897548 #68218]  INFO -- net.ssh.transport.server_version[e790]: negotiating protocol version
D, [2021-10-06T12:47:35.897561 #68218] DEBUG -- net.ssh.transport.server_version[e790]: local is `SSH-2.0-Ruby/Net::SSH_6.1.0 x86_64-darwin19'
D, [2021-10-06T12:47:35.915149 #68218] DEBUG -- net.ssh.transport.server_version[e790]: remote is `SSH-2.0-OpenSSH_8.8'
W, [2021-10-06T12:47:35.915263 #68218]  WARN -- net.ssh.transport.algorithms[e7a4]: unsupported algorithm: `["rsa-sha2-256", "rsa-sha2-512"]'
I, [2021-10-06T12:47:35.915400 #68218]  INFO -- net.ssh.transport.algorithms[e7a4]: sending KEXINIT
D, [2021-10-06T12:47:35.915592 #68218] DEBUG -- socket[e7b8]: queueing packet nr 0 type 20 len 796
D, [2021-10-06T12:47:35.915638 #68218] DEBUG -- socket[e7b8]: sent 800 bytes
D, [2021-10-06T12:47:35.915823 #68218] DEBUG -- socket[e7b8]: read 1048 bytes
D, [2021-10-06T12:47:35.915870 #68218] DEBUG -- socket[e7b8]: received packet nr 0 type 20 len 1044
I, [2021-10-06T12:47:35.915891 #68218]  INFO -- net.ssh.transport.algorithms[e7a4]: got KEXINIT from server
I, [2021-10-06T12:47:35.915941 #68218]  INFO -- net.ssh.transport.algorithms[e7a4]: negotiating algorithms
D, [2021-10-06T12:47:35.916109 #68218] DEBUG -- net.ssh.transport.algorithms[e7a4]: negotiated:
* kex: ecdh-sha2-nistp521
* host_key: ssh-ed25519
* encryption_server: aes256-ctr
* encryption_client: aes256-ctr
* hmac_client: hmac-sha2-512-etm@openssh.com
* hmac_server: hmac-sha2-512-etm@openssh.com
* compression_client: none
* compression_server: none
* language_client:
* language_server:
D, [2021-10-06T12:47:35.916125 #68218] DEBUG -- net.ssh.transport.algorithms[e7a4]: exchanging keys
D, [2021-10-06T12:47:35.919055 #68218] DEBUG -- socket[e7b8]: queueing packet nr 1 type 30 len 148
D, [2021-10-06T12:47:35.919091 #68218] DEBUG -- socket[e7b8]: sent 152 bytes
D, [2021-10-06T12:47:35.926237 #68218] DEBUG -- socket[e7b8]: read 312 bytes
D, [2021-10-06T12:47:35.926292 #68218] DEBUG -- socket[e7b8]: received packet nr 1 type 31 len 292
D, [2021-10-06T12:47:35.929370 #68218] DEBUG -- socket[e7b8]: queueing packet nr 2 type 21 len 20
D, [2021-10-06T12:47:35.929410 #68218] DEBUG -- socket[e7b8]: sent 24 bytes
D, [2021-10-06T12:47:35.929441 #68218] DEBUG -- socket[e7b8]: received packet nr 2 type 21 len 12
D, [2021-10-06T12:47:35.929801 #68218] DEBUG -- net.ssh.authentication.session[e7cc]: beginning authentication of `clear'
D, [2021-10-06T12:47:35.929833 #68218] DEBUG -- socket[e7b8]: using encrypt-then-mac
D, [2021-10-06T12:47:35.929875 #68218] DEBUG -- socket[e7b8]: queueing packet nr 3 type 5 len 32
D, [2021-10-06T12:47:35.929906 #68218] DEBUG -- socket[e7b8]: sent 100 bytes
D, [2021-10-06T12:47:35.930998 #68218] DEBUG -- socket[e7b8]: read 100 bytes
D, [2021-10-06T12:47:35.931064 #68218] DEBUG -- socket[e7b8]: received packet nr 3 type 6 len 32
D, [2021-10-06T12:47:35.931120 #68218] DEBUG -- net.ssh.authentication.session[e7cc]: trying none
D, [2021-10-06T12:47:35.931254 #68218] DEBUG -- socket[e7b8]: using encrypt-then-mac
D, [2021-10-06T12:47:35.931287 #68218] DEBUG -- socket[e7b8]: queueing packet nr 4 type 50 len 48
D, [2021-10-06T12:47:35.931318 #68218] DEBUG -- socket[e7b8]: sent 116 bytes
D, [2021-10-06T12:47:35.939375 #68218] DEBUG -- socket[e7b8]: read 116 bytes
D, [2021-10-06T12:47:35.939468 #68218] DEBUG -- socket[e7b8]: received packet nr 4 type 51 len 48
D, [2021-10-06T12:47:35.939515 #68218] DEBUG -- net.ssh.authentication.session[e7cc]: allowed methods: publickey,keyboard-interactive
D, [2021-10-06T12:47:35.939548 #68218] DEBUG -- net.ssh.authentication.methods.none[e7e0]: none failed
D, [2021-10-06T12:47:35.939575 #68218] DEBUG -- net.ssh.authentication.session[e7cc]: trying publickey
D, [2021-10-06T12:47:35.940158 #68218] DEBUG -- net.ssh.authentication.agent[e7f4]: connecting to ssh-agent
D, [2021-10-06T12:47:35.940245 #68218] DEBUG -- net.ssh.authentication.agent[e7f4]: sending agent request 1 len 47
D, [2021-10-06T12:47:35.940318 #68218] DEBUG -- net.ssh.authentication.agent[e7f4]: received agent packet 5 len 1
D, [2021-10-06T12:47:35.940345 #68218] DEBUG -- net.ssh.authentication.agent[e7f4]: sending agent request 11 len 0
D, [2021-10-06T12:47:35.940419 #68218] DEBUG -- net.ssh.authentication.agent[e7f4]: received agent packet 12 len 637
D, [2021-10-06T12:47:35.940625 #68218] DEBUG -- net.ssh.authentication.methods.publickey[e81c]: trying publickey (dd:3b:b8:2e:85:04:06:e9:ab:ff:a8:0a:c0:04:6e:d6)
D, [2021-10-06T12:47:35.940677 #68218] DEBUG -- socket[e7b8]: using encrypt-then-mac
D, [2021-10-06T12:47:35.940762 #68218] DEBUG -- socket[e7b8]: queueing packet nr 5 type 50 len 352
D, [2021-10-06T12:47:35.940802 #68218] DEBUG -- socket[e7b8]: sent 420 bytes
D, [2021-10-06T12:47:35.948355 #68218] DEBUG -- socket[e7b8]: read 116 bytes
D, [2021-10-06T12:47:35.948433 #68218] DEBUG -- socket[e7b8]: received packet nr 5 type 51 len 48
D, [2021-10-06T12:47:35.948469 #68218] DEBUG -- net.ssh.authentication.session[e7cc]: allowed methods: publickey,keyboard-interactive
E, [2021-10-06T12:47:35.948496 #68218] ERROR -- net.ssh.authentication.session[e7cc]: all authorization methods failed (tried none, publickey)

DEBUG ssh: == Net-SSH connection debug-level log END ==
 INFO ssh: SSH not ready: #<Vagrant::Errors::SSHAuthenticationFailed: SSH authentication failed! This is typically caused by the public/private
keypair for the SSH user not being properly set on the guest VM. Please
verify that the guest VM is setup with the proper public key, and that
the private key path for Vagrant is setup properly as well.>
 INFO subprocess: Starting process: ["/Applications/VMware Fusion.app/Contents/Library/vmrun", "list"]
DEBUG subprocess: Command not in installer, not touching env vars.
 INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: Total running VMs: 1
/Users/am/hacks/ClearLinux-packer/extras/libvirt.native/.vagrant/machines/libvirt/vmware_fusion/885a3a70-5bd1-475b-9fd2-911b85939a36/ClearLinux.vmx
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 31999
```

Vagrant host is a Mac running latest BigSur. 

Thanks in advance and keep the great work!

António

**Comment by chrisroberts** [comment]
Oh, awesome. Thanks @AntonioMeireles, I'll have a look at this today.

**Comment by chrisroberts** [comment]
Okay, connecting to an OpenSSH server running 8.8 appears to work correctly with Vagrant in its current state. Is there any way you can package up the guest you are using and toss it on Vagrant Cloud so I can pull it down and inspect the error you are encountering?

**Comment by AntonioMeireles** [comment]
@chrisroberts 

To reproduce, given bellow `Vagrantfile` ... 
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "AntonioMeireles/ClearLinux"
end
```
just do ...
- `vagrant up`
- `vagrant ssh sudo swupd update`
   > this will migrate VM to OpenSSH 8.8...
- `vagrant reload`
   > this should now fail...
   
   
Just bug me if you need anything else!

All the best, 

António

**Comment by chrisroberts** [comment]
@AntonioMeireles Are you encountering a failure during provisioning? I pulled the box, updated, reloaded, and ran a provision which was successful. Checking the ssh version on the guest, it shows as 8.8.

**Comment by AntonioMeireles** [comment]
(on mobile) yes, provisioning fails per the error posted initially.
(macos). plain ssh works. can try tomorrow (late here today) in different
setup.

**Comment by chrisroberts** [comment]
Okay, great. If you can provide a gist of the debug output from a failed provision that would be very helpful to identify if there's some other cause for the error. Thanks so much :slightly_smiling_face:

**Comment by AntonioMeireles** [comment]
(connected again) 

sorry. example above worked because `Vagrantfile` needs a provisioner to fail. _working_ (i.e failing) PoC and logs [here](https://gist.github.com/AntonioMeireles/92fd240c8e0a8532ade7d07b21931652)

Thanks once again for your patience!

**Comment by dentarg** [comment]
@chrisroberts You can try the box `twingly/openbsd-7.0-amd64` on Vagrant Cloud (see https://github.com/twingly/boxes/issues/33), there's output from `vagrant up` over there (and I can also share the debug output with you if still need that (it was a lot)) 

(note: the provision step does not fail for me (I use Ansible) and I can SSH to the box manually)

**Comment by chrisroberts** [comment]
@AntonioMeireles Would you try adding this to your Vagrantfile and see if you get different behavior:

```ruby
Vagrant.configure("2") do |config|
  # Existing configuration

  config.vm.provider :vmware_desktop do |v|
    v.ssh_info_public = true
  end
end
```

Thanks!

**Comment by chrisroberts** [comment]
@dentarg Would you please do a fresh `vagrant up --debug` and provide a gist of the output generated. As you described in your linked issue, after the `up` command times out you are still able to provision. So after the `up` times out, would you also run a `vagrant provision --debug` and provide a gist of that output as well.

Thanks so much!

**Comment by AntonioMeireles** [comment]
@chrisroberts 

still failing, as before. do you want the DEBUG logs again ?

**Comment by chrisroberts** [comment]
My assumption is that it's failing in the same way (continually waiting for the communicator to become ready), and if that's correct I don't need the debug logs. The behavior is odd given that I can run the same thing locally and see no issues.

I think something that might be useful is to do a `vagrant up` with the provider `gui` option enabled. Then log into the guest directly and see what sshd is logging about the connection attempts. That might (hopefully) provide some clues as to what's going on.

**Comment by AntonioMeireles** [comment]
@chrisroberts 

fwiw i don't need to enable gui mode to get in. plain ssh works in another window (always worked afaict). on VM's side in logs i only get tons of...
```
Nov 02 22:03:44 clearlinux sshd[3900]: userauth_pubkey: key type ssh-rsa not in PubkeyAcceptedAlgorithms [preauth]
Nov 02 22:03:44 clearlinux sshd[3900]: Connection closed by authenticating user clear 172.16.46.1 port 54743 [preauth]
```
for every attempt logging in via the bundled built-in net-ssh.  i should had taken a 'bit' more attention to that specific error **before** because ... well because adding `PubkeyAcceptedKeyTypes=+ssh-rsa` to VMs `sshd_config` made things suddently behave from net-ssh's side. 

so to summarize:
- this specific behaviour appeared here after guest OS  was migrated to openssh-8.8.
- openssh (client) never was affected
- internal ssh client from Vagrant (at least here) started failing. **FIX**/**WORKAROUND** seems to be to add `PubkeyAcceptedKeyTypes=+ssh-rsa` to VMs `/etc/ssh/sshd_config`

so, if this is perhaps still an issue - at least for me isn't anymore a blocker one, so you can drop this lower in your queue or just close this as there's an workaround. Thanks again for all your patience!

**Comment by dentarg** [comment]
Yeah, I posted what the server (sshd) logged in https://github.com/twingly/boxes/issues/33 and suspected that it would be possible to workaround by tweaking the sshd_config

**Comment by chrisroberts** [comment]
@dentarg Can you provide the debug gists from an `up` and `provision`? I'm trying to determine what's common and what's not where these commands are failing in this way. Adding the `+ssh-rsa` should not be required as Vagrant is providing patches to the net-ssh lib to support that signature type being deprecated. And locally for me, using either of your boxes "as-is", I can connect without issue using the net-ssh communicator. 

Thanks for the help in trying to track this down!

**Comment by webhead404** [comment]
I am having an interesting issue with this as well that doesn't directly tie to this but I do think has some issues with the net-ssh library. Using a CentOS box running an older version of openSSH. However during provisioning after they key steps Vagrant times out. Virtualbox 6.1.26 and Vagrant 2.2.18.

**Comment by dentarg** [comment]
@chrisroberts I've emailed you the debug logs (using the email on your GitHub profile) plus `/var/log/authlog` from the OpenBSD VM. I have a complete repro at https://github.com/dentarg/gists/tree/master/gists/vagrant-issue-12533, it reproduces for me using macOS 10.15.7, Vagrant 2.2.19 and VirtualBox 6.1.28r147628.

It actually worked one time, right after I had updated Vagrant from 2.2.14 to 2.2.19, but I didn't run it with `--debug` then:

```
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'twingly/openbsd-7.0-amd64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'twingly/openbsd-7.0-amd64' version '1.0.0' is up to date...
==> default: Setting the name of the VM: vagrant-issue-12533_default_1636410433183_10149
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
==> default: Running provisioner: ansible...
    default: Running ansible-playbook...

PLAY [default] *****************************************************************

TASK [raw] *********************************************************************
changed: [default]

PLAY RECAP *********************************************************************
default                    : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
```

**Comment by dentarg** [comment]
@chrisroberts I've installed both VirtualBox and Vagrant using Homebrew

**Comment by chrisroberts** [comment]
@dentarg If it's possible, can you replace the net-ssh patch file in your local install with the one in PR #12584 and see if that resolves the connection problem. It can be testable with `vagrant up` but also a `vagrant provision` if using a shell provisioner like:

```ruby
...
config.vm.provision :shell, inline: 'echo hi'
...
```

Thanks so much!

**Comment by dentarg** [comment]
@chrisroberts Still getting `Authentication failure` with the patch, I've emailed you some fresh debug logs

**Comment by dentarg** [comment]
Here's a reproducer outside of Vagrant

```ruby
require "bundler/inline"

gemfile do
  source "https://rubygems.org"
  gem "net-ssh", "6.1.0"
  gem "ed25519"
  gem "bcrypt_pbkdf"
end

class Vagrant
  class << self
    def global_logger
      self.new
    end
  end

  def method_missing(symbol, *args)
    puts "Vagrant.global_logger #{symbol.upcase} #{args}"
  end
end

# https://github.com/hashicorp/vagrant/pull/12584
require_relative "patches/net-ssh"

require "net/ssh"

ssh_info = {
  host: "127.0.0.1",
  username: "vagrant",
}
connect_opts = {
  port: 2222,
  verbose: :debug,
  keys: "/Users/dentarg/.vagrant.d/insecure_private_key",
  keys_only: true,
  auth_methods: ["publickey"]
}

Net::SSH.start(ssh_info[:host], ssh_info[:username], connect_opts)
```

Here's logs from macOS 10.15.7 using Ruby 2.7.4:

```
$ ruby repro.rb
D, [2021-11-12T01:27:11.946524 #85659] DEBUG -- net.ssh.transport.session[58c]: establishing connection to 127.0.0.1:2222
D, [2021-11-12T01:27:11.947120 #85659] DEBUG -- net.ssh.transport.session[58c]: connection established
I, [2021-11-12T01:27:11.947208 #85659]  INFO -- net.ssh.transport.server_version[5a0]: negotiating protocol version
D, [2021-11-12T01:27:11.947222 #85659] DEBUG -- net.ssh.transport.server_version[5a0]: local is `SSH-2.0-Ruby/Net::SSH_6.1.0 x86_64-darwin19'
D, [2021-11-12T01:27:11.971361 #85659] DEBUG -- net.ssh.transport.server_version[5a0]: remote is `SSH-2.0-OpenSSH_8.8'
I, [2021-11-12T01:27:11.972873 #85659]  INFO -- net.ssh.transport.algorithms[5b4]: sending KEXINIT
D, [2021-11-12T01:27:11.973000 #85659] DEBUG -- socket[5c8]: queueing packet nr 0 type 20 len 844
D, [2021-11-12T01:27:11.973080 #85659] DEBUG -- socket[5c8]: sent 848 bytes
D, [2021-11-12T01:27:11.973213 #85659] DEBUG -- socket[5c8]: read 1048 bytes
D, [2021-11-12T01:27:11.973263 #85659] DEBUG -- socket[5c8]: received packet nr 0 type 20 len 1044
I, [2021-11-12T01:27:11.973287 #85659]  INFO -- net.ssh.transport.algorithms[5b4]: got KEXINIT from server
I, [2021-11-12T01:27:11.973330 #85659]  INFO -- net.ssh.transport.algorithms[5b4]: negotiating algorithms
D, [2021-11-12T01:27:11.973397 #85659] DEBUG -- net.ssh.transport.algorithms[5b4]: negotiated:
* kex: ecdh-sha2-nistp521
* host_key: ecdsa-sha2-nistp256
* encryption_server: aes256-ctr
* encryption_client: aes256-ctr
* hmac_client: hmac-sha2-512-etm@openssh.com
* hmac_server: hmac-sha2-512-etm@openssh.com
* compression_client: none
* compression_server: none
* language_client:
* language_server:
D, [2021-11-12T01:27:11.973984 #85659] DEBUG -- net.ssh.transport.algorithms[5b4]: exchanging keys
D, [2021-11-12T01:27:11.976714 #85659] DEBUG -- socket[5c8]: queueing packet nr 1 type 30 len 148
D, [2021-11-12T01:27:11.976833 #85659] DEBUG -- socket[5c8]: sent 152 bytes
D, [2021-11-12T01:27:11.997016 #85659] DEBUG -- socket[5c8]: read 376 bytes
D, [2021-11-12T01:27:11.997115 #85659] DEBUG -- socket[5c8]: received packet nr 1 type 31 len 356
D, [2021-11-12T01:27:11.999908 #85659] DEBUG -- socket[5c8]: queueing packet nr 2 type 21 len 20
D, [2021-11-12T01:27:11.999959 #85659] DEBUG -- socket[5c8]: sent 24 bytes
D, [2021-11-12T01:27:12.000054 #85659] DEBUG -- socket[5c8]: received packet nr 2 type 21 len 12
Vagrant.global_logger DEBUG ["ssh server version detected: 8.8"]
Vagrant.global_logger DEBUG ["ssh server supports deprecation of RSA SHA1, deprecating"]
D, [2021-11-12T01:27:12.000337 #85659] DEBUG -- net.ssh.authentication.session[5dc]: beginning authentication of `vagrant'
D, [2021-11-12T01:27:12.000368 #85659] DEBUG -- socket[5c8]: using encrypt-then-mac
D, [2021-11-12T01:27:12.000417 #85659] DEBUG -- socket[5c8]: queueing packet nr 3 type 5 len 32
D, [2021-11-12T01:27:12.000445 #85659] DEBUG -- socket[5c8]: sent 100 bytes
D, [2021-11-12T01:27:12.000940 #85659] DEBUG -- socket[5c8]: read 100 bytes
D, [2021-11-12T01:27:12.001002 #85659] DEBUG -- socket[5c8]: received packet nr 3 type 6 len 32
D, [2021-11-12T01:27:12.001072 #85659] DEBUG -- net.ssh.authentication.session[5dc]: trying publickey
Vagrant.global_logger DEBUG ["set RSA SHA1 deprecation on public key: bb:e4:5a:cb:47:24:59:5e:27:4a:2a:5e:72:2b:06:61"]
Vagrant.global_logger DEBUG ["set RSA SHA1 deprecation on private key: bb:e4:5a:cb:47:24:59:5e:27:4a:2a:5e:72:2b:06:61"]
D, [2021-11-12T01:27:12.001892 #85659] DEBUG -- net.ssh.authentication.agent[5f0]: connecting to ssh-agent
D, [2021-11-12T01:27:12.002014 #85659] DEBUG -- net.ssh.authentication.agent[5f0]: sending agent request 1 len 47
D, [2021-11-12T01:27:12.002158 #85659] DEBUG -- net.ssh.authentication.agent[5f0]: received agent packet 5 len 1
D, [2021-11-12T01:27:12.002202 #85659] DEBUG -- net.ssh.authentication.agent[5f0]: sending agent request 11 len 0
D, [2021-11-12T01:27:12.002324 #85659] DEBUG -- net.ssh.authentication.agent[5f0]: received agent packet 12 len 1550
D, [2021-11-12T01:27:12.002648 #85659] DEBUG -- net.ssh.authentication.methods.publickey[618]: trying publickey (dd:3b:b8:2e:85:04:06:e9:ab:ff:a8:0a:c0:04:6e:d6)
D, [2021-11-12T01:27:12.002700 #85659] DEBUG -- socket[5c8]: using encrypt-then-mac
D, [2021-11-12T01:27:12.002739 #85659] DEBUG -- socket[5c8]: queueing packet nr 4 type 50 len 352
D, [2021-11-12T01:27:12.002798 #85659] DEBUG -- socket[5c8]: sent 420 bytes
D, [2021-11-12T01:27:12.019889 #85659] DEBUG -- socket[5c8]: read 132 bytes
D, [2021-11-12T01:27:12.020006 #85659] DEBUG -- socket[5c8]: received packet nr 4 type 51 len 64
D, [2021-11-12T01:27:12.020049 #85659] DEBUG -- net.ssh.authentication.session[5dc]: allowed methods: publickey,password,keyboard-interactive
E, [2021-11-12T01:27:12.020079 #85659] ERROR -- net.ssh.authentication.session[5dc]: all authorization methods failed (tried publickey)
Traceback (most recent call last):
  1: from repro.rb:41:in `<main>'
/Users/dentarg/.gem/ruby/2.7.4/gems/net-ssh-6.1.0/lib/net/ssh.rb:268:in `start': Authentication failed for user vagrant@127.0.0.1 (Net::SSH::AuthenticationFailed)
```

```
$ ssh-keygen -f /Users/dentarg/.vagrant.d/insecure_private_key -y | ssh-keygen -f - -l -E md5
2048 MD5:dd:3b:b8:2e:85:04:06:e9:ab:ff:a8:0a:c0:04:6e:d6 no comment (RSA)
```

Here's what's logged in OpenBSD when the above program is run:

```
openbsd70$ tail -n0 -f /var/log/authlog
Nov 12 00:27:11 openbsd70 sshd[38915]: userauth_pubkey: key type ssh-rsa not in PubkeyAcceptedAlgorithms [preauth]
Nov 12 00:27:11 openbsd70 sshd[38915]: Connection closed by authenticating user vagrant 10.0.2.2 port 55069 [preauth]
```

**Comment by webhead404** [comment]
I used the above info to test with CentOS 7 boxes (not running openssh-8.x+. However after replacing the patch file with #12584 Vagrant connects as expected!

macOS 11.6
Vagrant 2.2.19
VirtualBox 6.1.26

**Comment by dentarg** [comment]
> Here's a reproducer outside of Vagrant

Using no patches and the master branch of `net-ssh` (https://github.com/net-ssh/net-ssh/commit/a45f54fe1de434605af0b7195dd9a91bccd2cec5, the merge of https://github.com/net-ssh/net-ssh/pull/838)) it works: `net.ssh.authentication.methods.publickey[6f4]: publickey succeeded`

**Comment by webhead404** [comment]
@dentarg how did you avoid using the patches?

**Comment by dentarg** [comment]
I ran the script I posted in https://github.com/hashicorp/vagrant/issues/12533#issuecomment-966723492 (it does the same as Vagrant does to connect)
